### PR TITLE
[WIP] - Utilities to normalize responses

### PR DESCRIPTION
**Pull request by nicobes14**
# Description

Creation of the Utils folder with a series of functions that return objects to normalize the responses
### Changes to schema validation middleware to show a use case

> before

> after

### Changes to the category controller

> before (list category function example)

> after 

### Changes to the category service 

> before

> after

- In this case the lines of codes are the same but it requires a validation in the controller and throw new Error() does not return an object

## Evidence
- Schema check responses
image
- Create a category returns same responses before and after but with less code and the logic is in the utils
- If there is a problem creating the category in the database
before
image
after
image

### CODE REVIEW

File: services/categories.js [hunk 1]
```diff
@@ -1,31 +1,32 @@
 const db = require('../database/models')
+const { checkUndefined, databaseError, created } = require('../utils')
 
 const { Category } = db
 // todo - create a error handler
 const listCategoryById = async (id) => {
   try {
     const category = await Category.findByPk(id)
-    return category
+    return checkUndefined(category)
```
Comment (HernanGarcia1985) [inline services/categories.js:9]: Falta errorMessage

Comment (nicobes14) [reply services/categories.js:9]: Lo puse asi por que hay un mensaje por defecto en el notFound, pero si quisieras tambien podes pasarle un mensaje de error

Comment (HernanGarcia1985) [reply services/categories.js:9]: Genial! Tenes razón, se me pasó a mi!

**Comment by HernanGarcia1985** [comment]
Excelente Nico!

**Pull request closed without merging**